proto/backend.proto

```diff
@@ -0,0 +1,159 @@
+syntax = "proto3";
+package pluginv2;
+
+//-----------------------------------------------
+// Common
+//-----------------------------------------------
+
+message PluginConfig {
+  int64 id = 1;
+  int64 orgId = 2;
+  string name = 3;
+  string type = 4;
+  string url = 5;
+  string jsonData = 6;
+  map<string,string> decryptedSecureJsonData = 7;
+}
+
+message TimeRange {
+  string fromRaw = 1;
+  string toRaw = 2;
+  int64 fromEpochMs = 3;
+  int64 toEpochMs = 4;
+}
+
+
+//-----------------------------------------------
+// QUERY
+//-----------------------------------------------
+
+message QueryRequest {
+  // Plugin Configuration
+  PluginConfig config = 1;
+
+  // Environment info
+  map<string,string> headers = 2;
+
+  // List of queries
+  repeated DataQuery queries = 3;
+}
+
+message DataQuery {
+  string refId = 1;
+  int64 maxDataPoints = 2;
+  int64 intervalMs = 3;
+  TimeRange timeRange = 4; // NOTE: this moved from QueryRequest in v1
+  string json = 5; // Arbitrary structure (model?  modelJson in v1)
+}
+
+message QueryResponse {
+  // Arrow encoded DataFrames
+  // Each frame encodes its own: Errors, meta, and refId
+  repeated bytes data = 1;
+
+  // Additional response metadata
+  map<string,string> headers = 2;
+}
+
+
+//-----------------------------------------------
+// HTTP style request
+//-----------------------------------------------
+
+message PluginRequest {
+  // Plugin Configuration
+  PluginConfig config = 1;
```

Comment:
This currently sends the config with every request -- this is required for datasources since a single plugin has multiple configurations.  An alternative approach would be to:
1. send a `configId` -- maybe an unguessable string that uniquely identifies `plugin+instanceId+org+lastUpdateTime`
2. add a function that lets the backend ask for config on demand with the `configId` 

I'm not sure which model makes more sense -- for things like database connection, knowing when you have a new one vs an reusing an open pool seems nice, but for access to the 'url' maybe it is more trouble than its worth

Reply:
Probably good to take into consideration. I haven't had time to review this yet, but I probably have some opinion :smile: Haven't had time to read your doc yet either so please bare with me. But related to issue Grafana should handle most of the things so the plugin developer don't need to care about things like caching etc.

Reply:
#2 The host also needs a way of saying that the config has been updated. 

It would be nice to solve this in a better way but I'm not sure important it is.

Reply:
With `configId` the thought is that the ID would change when it is updated -- that way it knows to get a new one.  The problem is that the plugin never knows the old one has changed -- but that matches the current situation if a backend plugin tries to cache anything.